src/server/pps/server/master.go

```diff
@@ -198,7 +203,15 @@ func (m *ppsMaster) attemptStep(ctx context.Context, e *pipelineEvent) error {
 	})
 	// we've given up on the step, check if the error indicated that the pipeline should fail
 	if err != nil && errors.As(err, &stepErr) && stepErr.failPipeline {
-		failError := m.a.setPipelineFailure(ctx, e.pipeline, fmt.Sprintf(
+		var specCommit *pfs.Commit
+		if err := m.a.txnEnv.WithReadContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
+			var err error
+			specCommit, err = m.a.findPipelineSpecCommit(txnCtx, e.pipeline, "")
```

Comment:
It seems like it would be better to have a transactional and non-transactional version of this so each call site that doesn't need the transaction won't have to set one up.

Reply:
Sure, I'll add that